Add support for remmaping arguments

PerrineAguiar opened the pull request:
Fixes #288.

PerrineAguiar commented:
@jubeira @adamantivm FYI
I tested it but it seems that there is an error on the rosjava side, I opened a PR in the rosjava_core repo to fix this: https://github.com/rosjava/rosjava_core/pull/248.

adamantivm commented:
Thank you @PerrineAguiar ! Please let me know when releasing a new public version of rosjava core becomes blocking for this or a new TRS release.

mcopejans commented on TangoRosStreamer/tango_nodelet_manager/src/main/jni/jni.cc:
```diff
@@ -57,9 +57,31 @@ Java_eu_intermodalics_nodelet_1manager_TangoNodeletManager_execute(
+          env->ReleaseStringUTFChars(remap_to_value, remap_to);
+        }
+      } else {
+        LOG(WARNING) << "Wrong remapping arguments array, its size should be an even number but is "
```
Make this a LOG(ERROR) and return in this case. Motivation: if the array is not empty, the user wants to use the remapping feature. So it is not used correctly, this should be made clear.

mcopejans commented on TangoRosStreamer/tango_nodelet_manager/src/main/jni/jni.cc:
```diff
@@ -57,9 +57,31 @@ Java_eu_intermodalics_nodelet_1manager_TangoNodeletManager_execute(
+          << remappingStringCount << ". Remapping will not be apply.";
+      }
+    } else {
+      LOG(WARNING) << "Remapping arguments array is null. Remapping will not be apply.";
```
LOG(INFO) << No topic remapping to be done.
Do not put implementation details in the logging.

PerrineAguiar commented:
@mcopejans Thanks for your review. I addressed your comments and change the code to follow the [ROS remapping syntax](http://wiki.ros.org/Remapping%20Arguments).

mcopejans commented:
LGTM

PerrineAguiar commented:
This PR was waiting for [this fix](https://github.com/rosjava/rosjava_core/pull/248) on the rosjava side.
A [new version of rosjava_core](https://github.com/rosjava/rosjava_mvn_repo/pull/44) which includes the fix has been released.
After testing that the remapping feature works as expected, I am merging this PR.
